app/src/main/java/uk/gov/android/authentication/login/AuthenticationError.kt

```diff
@@ -2,24 +2,34 @@ package uk.gov.android.authentication.login
 
 import android.content.Intent
 import net.openid.appauth.AuthorizationException
+import net.openid.appauth.AuthorizationException.AuthorizationRequestErrors
 
 class AuthenticationError(
     override val message: String,
     val type: ErrorType
 ) : Error() {
     enum class ErrorType {
         OAUTH,
-        NETWORK
+        ACCESS_DENIED
     }
 
     companion object {
         internal const val NULL_AUTH_MESSAGE = "Auth response was null"
 
         fun from(intent: Intent) = from(AuthorizationException.fromIntent(intent))
 
-        fun from(exception: AuthorizationException?) = AuthenticationError(
-            message = exception?.message ?: NULL_AUTH_MESSAGE,
-            type = ErrorType.OAUTH
-        )
+        fun from(exception: AuthorizationException?): AuthenticationError {
+            return when (exception) {
+                AuthorizationRequestErrors.ACCESS_DENIED -> AuthenticationError(
+                    message = exception?.message ?: NULL_AUTH_MESSAGE,
```

Comment:
not sure you need the null check
```suggestion
                    message = exception.message ?: NULL_AUTH_MESSAGE,
```

Reply:
unfortunately, it doesn't work, it throws an error as the exception is could be null, but the `NULL_AUTH_MESSAGE` is defaulting only for the message, not the exception itself - does that make sense? not sure how to explain it

Reply:
all done, thank you